protocol/messenger_activity_center.go

```diff
@@ -115,38 +115,48 @@ func (m *Messenger) MarkAsSeenActivityCenterNotifications() (*MessengerResponse,
 }
 
 func (m *Messenger) MarkAllActivityCenterNotificationsRead(ctx context.Context) (*MessengerResponse, error) {
-	response := &MessengerResponse{}
 	updateAt := m.GetCurrentTimeInMillis()
-	if m.hasPairedDevices() {
-		ids, err := m.persistence.GetNotReadActivityCenterNotificationIds()
-		if err != nil {
-			return nil, err
-		}
 
-		_, err = m.MarkActivityCenterNotificationsRead(ctx, toHexBytes(ids), updateAt, true)
+	ids, err := m.persistence.GetNotReadActivityCenterNotificationIds()
+	if err != nil {
 		return nil, err
 	}
 
-	err := m.persistence.MarkAllActivityCenterNotificationsRead(updateAt)
+	return m.MarkActivityCenterNotificationsRead(ctx, toHexBytes(ids), updateAt, true)
+}
+
+func (m *Messenger) MarkActivityCenterNotificationsRead(ctx context.Context, ids []types.HexBytes, updatedAt uint64, sync bool) (*MessengerResponse, error) {
+	notifications, err := m.persistence.GetActivityCenterNotificationsByID(ids)
 	if err != nil {
 		return nil, err
 	}
 
-	state, err := m.persistence.GetActivityCenterState()
-	if err != nil {
-		return nil, err
+	response := &MessengerResponse{}
+	repliesAndMentions := make(map[string][]string)
+
+	// When marking as read Mention or Reply notification, the corresponding chat message should also be read.
+	for _, notification := range notifications {
+		notification.Read = true
+		response.AddActivityCenterNotification(notification)
+
+		if notification.Type == ActivityCenterNotificationTypeMention || notification.Type == ActivityCenterNotificationTypeReply {
+			repliesAndMentions[notification.ChatID] = append(repliesAndMentions[notification.ChatID], notification.ID.String())
```

Comment:
Would it be more correct to append `notification.Message.ID` instead of `notification.ID`? I'm asking this because at the end of the line we're updating the `user_messages` table.

Reply:
fixed